be/src/exprs/in_const_predicate.hpp

```diff
@@ -395,6 +399,120 @@ class VectorizedInConstPredicate final : public Predicate {
     std::vector<ColumnPtr> _string_values;
 };
 
+class VectorizedInConstPredicateGeneric final : public Predicate {
+public:
+    VectorizedInConstPredicateGeneric(const TExprNode& node)
+            : Predicate(node), _is_not_in(node.in_predicate.is_not_in) {}
+
+    VectorizedInConstPredicateGeneric(const VectorizedInConstPredicateGeneric& other)
+            : Predicate(other), _is_not_in(other._is_not_in) {}
+
+    ~VectorizedInConstPredicateGeneric() override = default;
+
+    Expr* clone(ObjectPool* pool) const override { return pool->add(new VectorizedInConstPredicateGeneric(*this)); }
+
+    Status open(RuntimeState* state, ExprContext* context, FunctionContext::FunctionStateScope scope) override {
+        RETURN_IF_ERROR(Expr::open(state, context, scope));
+        _const_input.resize(_children.size());
+        for (auto i = 0; i < _children.size(); ++i) {
+            if (dynamic_cast<VectorizedLiteral*>(_children[i])) {
+                ASSIGN_OR_RETURN(_const_input[i], _children[i]->evaluate_checked(context, nullptr));
+            } else {
+                _const_input[i] = nullptr;
+            }
+        }
+        return Status::OK();
+    }
+
+    StatusOr<ColumnPtr> evaluate_checked(ExprContext* context, Chunk* ptr) override {
+        auto child_size = _children.size();
+        Columns input_data(child_size);
+        std::vector<NullColumnPtr> input_null(child_size);
+        std::vector<bool> is_const(child_size);
+        Columns columns_ref(child_size);
+        ColumnPtr value;
+        bool all_const = true;
+        for (int i = 0; i < child_size; ++i) {
+            value = _const_input[i];
+            if (value == nullptr) {
+                ASSIGN_OR_RETURN(value, _children[i]->evaluate_checked(context, ptr));
```

Comment:
the children of `in` must be constant, I think you can add DCHECK on `expr.is_const`,  and don't need complex handle